rq/worker.py

```diff
@@ -698,6 +698,7 @@ def perform_job(self, job, queue):
         try:
             with self.death_penalty_class(job.timeout or self.queue_class.DEFAULT_TIMEOUT):
                 rv = job.perform()
+            job.refresh()
```

Comment:
That could work. It should probably be explicitly documented somewhere that the "meta" dict should never be modified by a Worker (or its subclasses), and that any other Job attributes (i.e., is_finished, return_value, etc.) should never be modified by any python code executed by the Job.

It would probably be useful in figure out a way to enforce those separation of concerns in the future, but I would settle for just this solution and some documentation instead. I can write up the function, where should the documentation go?